tests/base.py

```diff
@@ -0,0 +1,177 @@
+import tap_tester.connections as connections
+import tap_tester.menagerie as menagerie
+import tap_tester.runner as runner
+import os
+import unittest
+from datetime import datetime as dt
+import time
+
+class ExactTargetBase(unittest.TestCase):
+    START_DATE = ""
+    DATETIME_FMT = {
+        "%Y-%m-%dT%H:%M:%SZ",
+        "%Y-%m-%d %H:%M:%S",
+        "%Y-%m-%dT%H:%M:%S.%fZ"
+    }
+
+    def name(self):
+        return "tap_tester_exacttarget_base"
+
+    def tap_name(self):
+        return "tap-exacttarget"
+
+    def setUp(self):
+        required_env = {
+            "TAP_EXACTTARGET_CLIENT_ID",
+            "TAP_EXACTTARGET_CLIENT_SECRET",
+            "TAP_EXACTTARGET_TENANT_SUBDOMAIN",
+            "TAP_EXACTTARGET_V2_CLIENT_ID",
+            "TAP_EXACTTARGET_V2_CLIENT_SECRET",
+            "TAP_EXACTTARGET_V2_TENANT_SUBDOMAIN",
+        }
+        missing_envs = [v for v in required_env if not os.getenv(v)]
+        if missing_envs:
+            raise Exception("set " + ", ".join(missing_envs))
+
+    def get_type(self):
+        return "platform.exacttarget"
+
+    def get_credentials(self):
+        return {
+            'client_secret': os.getenv('TAP_EXACTTARGET_CLIENT_SECRET')
+        }
+
+    def get_properties(self, original: bool = True):
+        return_value = {
+            'start_date': '2019-01-01T00:00:00Z',
+            'client_id': os.getenv('TAP_EXACTTARGET_CLIENT_ID'),
+            'tenant_subdomain': os.getenv('TAP_EXACTTARGET_TENANT_SUBDOMAIN')
+        }
+        if original:
+            return return_value
+
+        # Reassign start date
+        return_value["start_date"] = self.START_DATE
+        return return_value
+
+    def expected_metadata(self):
+        # Note: Custom streams failed on our account with an error on
+        # `_CustomObjectKey` not being valid
+        return {
+            "campaign": {
+                "primary-key": {"id"},
+                "replication": "FULL"
```

Comment:
Instead of using `primary-key`, `replication`, `replication-key`, `FULL`, `INCREMENTAL`, create constants of those and use those. use FULL_TABLE instead of FULL

Reply:
Updated the code.